File: test_data/code/java/1566-detect-pattern-of-length-m-repeated-k-or-more-times.java
class Solution {
    public boolean containsPattern(int[] arr, int m, int k) {
        if (arr.length < m * k) return false;
        
        for (int i = 0; i <= arr.length - m * k; i++) {
            if (i + m - 1 < arr.length && Arrays.equals(Arrays.copyOfRange(arr, i, i + m), Arrays.copyOfRange(arr, i + m, i + m * k))) {
                return true;
            }
        }
        return false;
    }
}

File: test_data/code/python/1566-detect-pattern-of-length-m-repeated-k-or-more-times.py
Python
class Solution:
    def containsPattern(self, arr: List[int], m: int, k: int) -> bool:
        for i in range(len(arr) - m * k + 1):
            pattern = arr[i:i+m]
            for j in range(k-1):
                if arr[i+j*m:i+(j+1)*m] != pattern:
                    break
            else:
                return True
        return False